### PR TITLE
Simplify extension building

### PR DESCRIPTION
**Pull request by twsouthwick**
This change reduces the number of msbuild calls that need to occur so
that the CLI can just use the output of already built assemblies to grab
as extensions. This should help with performance as well as incremental
builds.

### CODE REVIEW

**Comment by sunandabalu** [comment]
Looks like the PR build is still failing @twsouthwick

**Comment by twsouthwick** [comment]
@sunandabalu That error was a transient one downloading the nuget looseassembly data file. I reorganized the targets to be smaller and clearer to understand as well. Once approved, I'll merge it in if you don't have other concerns.

**Review by sunandabalu — APPROVED** [review]

**Pull request merged**
